packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -123,9 +123,10 @@ export function findAssociatedDatasets(
 ): (NumArrayDataset | undefined)[] {
   const dsetList = attrValuesStore.getSingle(group, type) || [];
   const dsetNames = typeof dsetList === 'string' ? [dsetList] : dsetList;
-  assertArray<string>(dsetNames);
+  assertArray(dsetNames);
 
   return dsetNames.map((name) => {
+    assertStr(name);
```

Comment:
We were type-casting via a generic, here, so if the array contained anything else than `string` we may have landed on a bug that would have been hard to track down.